Bookish/Bookish.DataAccess/BarcodeService.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Drawing.Imaging;
+using System.IO;
+using System.Linq;
+using System.Text;
+using BarcodeLib;
+
+namespace Bookish.DataAccess
+{
+    public interface IBarcodeService
+    {
+        IEnumerable<int> GetBarcodeIds(string isbn, int copies, string imageFolder);
+    }
+    
+    public class BarcodeService : IBarcodeService
+    {
+        private readonly ILibraryService libraryService;
+        public BarcodeService(ILibraryService libraryService)
+        {
+            this.libraryService = libraryService;
+        }
+
+        public IEnumerable<int> GetBarcodeIds(string isbn, int copies, string imageFolder)
+        {
+            try
+            {
+                Clean(imageFolder);
+            }
+            catch
+            {
+                // ignored
+            }
+
+            var newCopyIds = libraryService.GetNewCopyIds(isbn, copies).ToList();
+            foreach (var copyId in newCopyIds)
+            {
+                CreateBarcodeImage(copyId, imageFolder);
+            }
+
+            return newCopyIds;
+        }
+
+        private static void CreateBarcodeImage(int copyId, string imageFolder)
+        {
+            var path = Path.Combine(imageFolder, $"barcode{copyId}.png");
+            if (File.Exists(path)) return;
```

Comment:
If the file exists you probably just want to overwrite it? 🙂

Reply:
The issue I was having was if someone refreshed the page it would try to overwrite the images and I'd get an error something like "Cannot overwrite as file is being used"

Reply:
Ah I see! I'd add a comment to explain that here 🙂